linkcheck/htmlutil/linkparse.py

```diff
@@ -82,7 +82,7 @@
 refresh_re = re.compile(r"(?i)^\d+;\s*url=(?P<url>.+)$")
 _quoted_pat = r"('[^']+'|\"[^\"]+\"|[^\)\s]+)"
 css_url_re = re.compile(r"url\(\s*(?P<url>%s)\s*\)" % _quoted_pat)
-swf_url_re = re.compile("(?i)%s" % urlutil.safe_url_pattern)
+swf_url_re = re.compile(b"(?i)%s" % urlutil.safe_url_pattern.encode('ascii'))
```

Comment:
could we get a clarification on why this encoding is necessary for swf and not other special snowflakes we have around here? :) "of course" isn't exactly a great one... ;)

Reply:
SWF files are binary, and so we use a byte string regex to grep a byte string with the file contents.

The .encode() is needed to convert an str regex to a byte string regex, so the regex compiler will know it'll be used to match byte strings.

> Both patterns and strings to be searched can be Unicode strings (str) as well as 8-bit strings (bytes). However, Unicode strings and 8-bit strings cannot be mixed: that is, you cannot match a Unicode string with a byte pattern or vice-versa; similarly, when asking for a substitution, the replacement string must be of the same type as both the pattern and the search string.

-- https://docs.python.org/3/library/re.html

Reply:
okay, makes sense. i'd still argue it would make sense to point that out in the comment here - it might not be obvious to our future selves wtf SWF is and why this matter. that and squash the commits and i will be happy(er).